datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -0,0 +1,558 @@
+						Description: "Notification targets for each rule case.",
+						Elem:        &schema.Schema{Type: schema.TypeString},
+					},
+					"status": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Severity of the Security Signal.",
+					},
+				},
+			},
+		},
+
+		"enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Whether the rule is enabled.",
+		},
+
+		"disabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Whether the rule is disabled.",
+		},
+
+		"message": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Message for generated signals.",
+		},
+
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The name of the rule.",
+		},
+
+		"options": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			MaxItems:    1,
+			Description: "Options on rules.",
+
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"evaluation_window": {
+						Type:        schema.TypeInt,
+						Required:    true,
+						Description: "A time window is specified to match when at least one of the cases matches true. This is a sliding window and evaluates in real time.",
+					},
+
+					"keep_alive": {
+						Type:        schema.TypeInt,
+						Required:    true,
+						Description: "Once a signal is generated, the signal will remain “open” if a case is matched at least once within this keep alive window.",
+					},
+
+					"max_signal_duration": {
+						Type:        schema.TypeInt,
+						Required:    true,
+						Description: "A signal will “close” regardless of the query being matched once the time exceeds the maximum duration. This time is calculated from the first seen timestamp.",
+					},
+				},
+			},
+		},
+
+		"query": {
+			Type:        schema.TypeList,
+			Required:    true,
+			Description: "Queries for selecting logs which are part of the rule.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"aggregation": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "The aggregation type.",
+					},
+					"distinct_fields": {
+						Type:        schema.TypeList,
+						Optional:    true,
+						Description: "Field for which the cardinality is measured. Sent as an array.",
+						Elem:        &schema.Schema{Type: schema.TypeString},
+					},
+					"group_by_fields": {
+						Type:        schema.TypeList,
+						Optional:    true,
+						Description: "Fields to group by.",
+						Elem:        &schema.Schema{Type: schema.TypeString},
+					},
+					"metric": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "The target field to aggregate over when using the sum or max aggregations.",
+					},
+					"name": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "Name of the query.",
+					},
+					"query": {
```

Comment:
`q` may be closer to the existing terraform API. I mostly used the same field names as the public API.